ios/browser/api/url/url_utils.h

```diff
@@ -53,6 +53,26 @@ OBJC_EXPORT
 
 /// Returns true if the URL has a scheme.
 - (bool)brave_hasScheme:(NSString*)scheme NS_SWIFT_NAME(hasScheme(scheme:));
+
+/**
+ This function will return a new url stripping any tracking query params.
+ If nothing is to be stripped, a null value is returned.
+
+ @param requestURL Indicates where we are navigating to.
+ @param initiatorURL Indicates the origin initiating the resource request. It's
+ the first url in a redirect chain.
+ @param redirectSourceURL Indicates the we are currently navigating from. It's
+ the last url in a redirect chain
+ @param method The HTTP method of the request.
+ @param isInternalRedirect Indicating if this is an internal redirect or not.
+ @return The url we should redirect to.
+ */
+- (nullable NSURL*)
+    brave_stripTrackerParamsUsingInitiatorURL:(NSURL*)initiatorURL
```

Comment:
also in general I think we should try to keep consistent directories so this should ideally be in ios/browser/query_filter/utils.h I think. Putting things in a different place in ios just makes the code harder to understand imo

Reply:
The `brave` prefix on these methods are because these are defined on a `NSURL` category, and Obj-C doesn't have name-spacing, so its possible to have duplicate symbols/method names for types we don't own. It was a typical thing to do in Obj-C back in the day (ref: https://developer.apple.com/library/archive/qa/qa1908/_index.html)

Reply:
sorry, actually ideally `components/query_filter/util` would go in `components/query_filter/core/util` and the obj-c interface would go in `components/query_filter/ios`
Can we do that in a follow-up if not here?

Reply:
I agree on the issue around inconsistent naming between C++ and Obj-C though

Reply:
@bridiver We can start doing layered components like that instead of having Obj-C interfaces live in `ios/` subdir but we'd still need a config of some sort to ensure those headers are copied into the framework. As of now its at least consistent in the way that `ios/browser/api` is setup. May also have to consider CODEOWNER stuff

Reply:
I forgot about the header issue, but let's discuss in slack because I don't think it should be a major problem to solve

Reply:
I think this should at least go in api/query_filter instead of url though

Reply:
@cuba for ☝️ we can still have it in a NSURL category, when Brandon originally mentioned making it category it didnt necessarily mean it needed to live in `api/url/url_utils.h`

Reply:
sounds good, I will move this back to the original folder in the following structure:

api/query_filter/url_utils.h/mm